src/effects/VST3/VST3Wrapper.cpp

```diff
@@ -416,6 +442,66 @@ VST3Wrapper::VST3Wrapper(VST3::Hosting::Module& module, const VST3::Hosting::Cla
    mEditController = editController;
 
    mEditController->initialize(&AudacityVst3HostApplication::Get());
+
+   if(GetCache(mEffectClassInfo.ID()) == nullptr)
```

Comment:
To make my last suggestion work, defaulted `VST3PluginCache` must be distinguishable from one that was initialized as in the following code, so you can write another test here.

Reply:
Yes, and all other places should check that too. Not obvious what should happen if `GetCache` returns not initialized object